src/Tribe/RSVP.php

```diff
@@ -2018,60 +2362,19 @@ public function generate_tickets_for( $product_id, $ticket_qty, $attendee_detail
 
 		// Iterate over all the amount of tickets purchased (for this product)
 		for ( $i = 0; $i < $qty; $i++ ) {
-
-			$attendee           = array(
-				'post_status' => 'publish',
-				'post_title'  => $attendee_full_name . ' | ' . ( $i + 1 ),
-				'post_type'   => self::ATTENDEE_OBJECT,
-				'ping_status' => 'closed',
-			);
-
-			// Insert individual ticket purchased
-			$attendee_id = wp_insert_post( $attendee );
-
-			if ( $status_stock_size > 0 ) {
-				// adjust total sales
-				$sales = (int) get_post_meta( $product_id, 'total_sales', true );
-				update_post_meta( $product_id, 'total_sales', ++ $sales );
-
-				// adjust stock
-				$stock = (int) get_post_meta( $product_id, '_stock', true ) - $status_stock_size;
-				update_post_meta( $product_id, '_stock', $stock );
+			try {
+				$this->create_attendee_for_ticket( $ticket_type, [
+					'full_name'         => $attendee_full_name,
+					'email'             => $attendee_email,
+					'optout'            => $attendee_optout,
+					'order_status'      => $attendee_order_status,
+					'order_id'          => $order_id,
+					'order_attendee_id' => $i + 1,
+					'user_id'           => is_user_logged_in() ? get_current_user_id() : 0,
+				] );
```

Comment:
:no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: a9445d604a15f08875b1cdde19ceca19</sub>